test/ReactWrapper-spec.js

```diff
@@ -369,12 +369,13 @@ describeWithDOM('mount', () => {
     it('should support data prop selectors', () => {
       const wrapper = mount(
         <div>
-          <span data-foo="bar" />
+          <span data-foo="bar" data-foo123="bar" />
         </div>
       );
 
       expect(wrapper.find('[data-foo="bar"]')).to.have.length(1);
       expect(wrapper.find('[data-foo]')).to.have.length(1);
+      expect(wrapper.find('[data-foo123]')).to.have.length(1);
```

Comment:
can you add another test for attributes that start with a digit? I'm pretty sure `123-abc` isn't a valid attribute (but either way we should have a test for it)